crates/cxx-qt-lib/src/core/qurl.rs

```diff
@@ -6,16 +6,159 @@ use cxx::{type_id, ExternType};
+        type QStringList = crate::QStringList;
+        include!("cxx-qt-lib/qurl.h");
         type QUrl = super::QUrl;
+
+        /// Resets the content of the QUrl. After calling this function,
+        /// the QUrl is equal to one that has been constructed with the default empty constructor.
+        fn clear(self: &mut QUrl);
+
+        /// Returns an error message if the last operation that modified this QUrl object ran into a parsing error.
+        /// If no error was detected, this function returns an empty string and isValid() returns true.
+        #[rust_name = "error_string"]
+        fn errorString(self: &QUrl) -> QString;
+
+        /// Returns true if this URL contains a fragment (i.e., if # was seen on it).
+        #[rust_name = "has_fragment"]
+        fn hasFragment(self: &QUrl) -> bool;
+
+        /// Returns true if this URL contains a Query (i.e., if ? was seen on it).
+        #[rust_name = "has_query"]
+        fn hasQuery(self: &QUrl) -> bool;
+
+        /// Returns true if the URL has no data; otherwise returns false.
+        #[rust_name = "is_empty"]
+        fn isEmpty(self: &QUrl) -> bool;
+
+        /// Returns true if this URL is pointing to a local file path. A URL is a local file path if the scheme is "file".
+        #[rust_name = "is_local_file"]
+        fn isLocalFile(self: &QUrl) -> bool;
+
+        /// Returns true if this URL is a parent of child_url.
+        /// childUrl is a child of this URL if the two URLs share the same scheme and authority,
+        /// and this URL's path is a parent of the path of child_url.
+        #[rust_name = "is_parent_of"]
+        fn isParentOf(self: &QUrl, child_url: &QUrl) -> bool;
+
+        /// Returns true if the URL is relative; otherwise returns false.
+        /// A URL is relative reference if its scheme is undefined;
+        /// this function is therefore equivalent to calling scheme().is_empty().
+        #[rust_name = "is_relative"]
+        fn isRelative(self: &QUrl) -> bool;
+
+        /// Returns true if the URL is non-empty and valid; otherwise returns false.
+        #[rust_name = "is_valid"]
+        fn isValid(self: &QUrl) -> bool;
+
+        /// Returns the port of the URL, or defaultPort if the port is unspecified.
+        #[rust_name = "port_or"]
+        fn port(self: &QUrl, port: i32) -> i32;
+
+        /// Returns the result of the merge of this URL with relative. This URL is used as a base to convert relative to an absolute URL.
+        fn resolved(self: &QUrl, relative: &QUrl) -> QUrl;
+
+        /// Returns the scheme of the URL. If an empty string is returned,
+        /// this means the scheme is undefined and the URL is then relative.
+        ///
+        /// The scheme can only contain US-ASCII letters or digits,
+        /// which means it cannot contain any character that would otherwise require encoding
+        /// Additionally, schemes are always returned in lowercase form.
+        fn scheme(self: &QUrl) -> QString;
```

Comment:
Maybe also wrap with `scheme_or_default`?

Reply:
Done and used the is_empty on the scheme to add a Option<T>